### DISCUSSION TITLE
Brew Upgrade SHA256 mismatch

### DISCUSSION THREAD

**Discussion by jschwanz** [JanDeDobbeleer/oh-my-posh#3425] [Troubleshoot] [answered]
I'm on a Mac running Ventura 13.1, and have oh-my-posh installed with Homebrew. I'm trying to upgrade from 13.9.0 to 14.0.0, but I'm getting a SHA256 mismatch when I do that.

```
==> Fetching jandedobbeleer/oh-my-posh/oh-my-posh
==> Downloading https://github.com/JanDeDobbeleer/oh-my-posh/archive/v14.0.0.tar.gz
==> Downloading from https://codeload.github.com/JanDeDobbeleer/oh-my-posh/tar.gz/refs/tags/v14.0.0
######################################################################## 100.0%
Error: SHA256 mismatch
Expected: 5b470f0b1678ff51435c167baa689e8bbef7f17b81cda2e1e9834ad3893696d0
  Actual: 253497b053e22d008738a3f13f5441f7374e61dc30ef0e1a2dacfbd61e5132d2
    File: /Users/myuser/Library/Caches/Homebrew/downloads/7f3ded000afbbc5d4bf061e36429e81ebe5aa64ea079f89a5970aead2429686a--oh-my-posh-14.0.0.tar.gz
```

**Comment by Bocephus** [comment]
I just ran into the same issue myself, only on a fresh install, and not an upgrade. Not that it matters much.

**Comment by jschwanz** [comment, marked as answer]
I just saw [this announcement (Git archive checksums may change) from GitHub](https://github.blog/changelog/2023-01-30-git-archive-checksums-may-change/), could that be related?

**Reply by JanDeDobbeleer** [reply]
@jschwanz they just changed that to revert. I'm going to validate this right now.

**Reply by JanDeDobbeleer** [reply]
@jschwanz @Bocephus seems to work again now, definitely related to that change and subsequent revert.

**Reply by ristaloff** [reply]
Had the same issue yesterday, can confirm it works now!

**Reply by JanDeDobbeleer** [reply]
I reached out to GitHub for a clarification and possible next steps. Because if they indeed plan to have variable SHA256 results for those zip files, I need to create a specific source zip with a constant hash myself.

**Reply by Bocephus** [reply]
Well... I am no longer getting the SHA256 error. I guess that is a plus. But now brew is telling me it must be built from source.

`==> Installing oh-my-posh from jandedobbeleer/oh-my-posh
Error: The following formula cannot be installed from bottle and must be
built from source.
  oh-my-posh
Install Clang or run 'brew install gcc'`

**Reply by Bocephus** [reply]
Looks like I'm good now after following the fix from [#1399](https://github.com/JanDeDobbeleer/oh-my-posh/issues/1339). I guess this has been added to the documentation, but I couldn't find it, and in my case, I was installing on Ubuntu 22.04 desktop, and not on a WSL2 instance, though I have it there as well, and it works fine.

I'll just chalk this up to me not RTFM :)

Thanks for the help.

**Reply by JanDeDobbeleer** [reply]
@Bocephus this is mentioned by homebrew as indicated [here](https://github.com/JanDeDobbeleer/oh-my-posh/issues/1339#issuecomment-980790753). And we also have this small remark in the [docs](https://ohmyposh.dev/docs/installation/linux):

<img width="941" alt="image" src="https://user-images.githubusercontent.com/2492783/215771851-2571bed6-f402-465c-b5a0-cf5f7d6a7994.png">

**Reply by jschwanz** [reply]
All looks good for me, too. Thanks.
